### ISSUE TITLE
not able to `gulp upgrade` a 7.1 theme to 7.2

### ISSUE THREAD

**Issue by john-co** [liferay/liferay-js-themes-toolkit#257]
**Steps to reproduce:**
1. create a theme using `yo liferay-theme`
2. run `gulp upgrade` on theme
3. Assert console error:
```
/home/john-co/jc-70-theme/node_modules/liferay-theme-tasks/lib/doctor.js:85
					throw new Error(
					^

Error: Task 'upgrade' is not supported for themes with version '7.1' in this version of 'liferay-theme-tasks'
    at assertTasksSupported (/home/john-co/jc-70-theme/node_modules/liferay-theme-tasks/lib/doctor.js:85:12)
    at doctor (/home/john-co/jc-70-theme/node_modules/liferay-theme-tasks/lib/doctor.js:43:2)
    at register (/home/john-co/jc-70-theme/node_modules/liferay-theme-tasks/index.js:62:2)
    at /home/john-co/jc-70-theme/node_modules/liferay-plugin-node-tasks/index.js:37:4
    at arrayEach (/home/john-co/jc-70-theme/node_modules/liferay-plugin-node-tasks/node_modules/lodash/lodash.js:516:11)
    at Function.forEach (/home/john-co/jc-70-theme/node_modules/liferay-plugin-node-tasks/node_modules/lodash/lodash.js:9344:14)
    at Object.module.exports.registerTasks (/home/john-co/jc-70-theme/node_modules/liferay-plugin-node-tasks/index.js:36:5)
    at Object.module.exports.registerTasks (/home/john-co/jc-70-theme/node_modules/liferay-theme-tasks/index.js:23:21)
    at Object.<anonymous> (/home/john-co/jc-70-theme/gulpfile.js:12:19)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Liftoff.handleArguments (/home/linuxbrew/.linuxbrew/lib/node_modules/gulp/bin/gulp.js:116:3)
    at Liftoff.execute (/home/linuxbrew/.linuxbrew/lib/node_modules/gulp/node_modules/liftoff/index.js:203:12)
    at module.exports (/home/linuxbrew/.linuxbrew/lib/node_modules/gulp/node_modules/flagged-respawn/index.js:51:3)
    at Liftoff.<anonymous> (/home/linuxbrew/.linuxbrew/lib/node_modules/gulp/node_modules/liftoff/index.js:195:5)
    at /home/linuxbrew/.linuxbrew/lib/node_modules/gulp/node_modules/liftoff/index.js:165:9
    at /home/linuxbrew/.linuxbrew/lib/node_modules/gulp/node_modules/v8flags/index.js:110:14
    at /home/linuxbrew/.linuxbrew/lib/node_modules/gulp/node_modules/v8flags/index.js:38:12
    at /home/linuxbrew/.linuxbrew/lib/node_modules/gulp/node_modules/v8flags/index.js:49:7
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
    at Function.Module.runMain (module.js:695:11)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
```

**Comment by wincent** [comment]
@john-co: this is with v8 of the toolkit, is that right?

As far as I know, the tool has never supported upgrading from 7.1 (for example, see [this array from an older release here](https://github.com/liferay/liferay-js-themes-toolkit/blob/d5489bf21e9acd426ac16eac444950c4f98ccad1/packages/liferay-theme-tasks/lib/doctor.js#L10-L11) that says you can upgrade from 6.2 or from 7.0, but nothing else).

v8.x of the themes toolkit doesn't provide any functionality related to DXP 7.2 (that includes functionality related to upgrading to 7.2). v9.x is the one for that. We have messages [like this one](https://github.com/liferay/liferay-js-themes-toolkit/blob/744343474eb3fd75c00ce3ef9d99cd72900797e0/packages/generator-liferay-theme/generators/common/messages.js#L14-L20) that tell them when the version of the toolkit they are using is too new, but we don't have inverse messages telling them when the version they are using is too old. I think it would be pretty straightforward to add them though, based on the change I made the other day in 4918edfe23318074c4c59f6649df88d12e9b.

So I think we should improve the error message here; it should instead say something like `Task 'upgrade' is not supported for themes with version '7.1' in this version of 'liferay-theme-tasks' - to upgrade a theme to version '7.2', please update to liferay-theme-tasks@^9.0.0-alpha.1 or newer`.

**Comment by john-co** [comment]
@wincent No I tried it on the v9.x alpha toolkit.

Sorry, I didn't include a step that says `uninstall v8.x and install v9.x alpha` between building a theme and upgrading. Is there a more proper way of upgrading a 7.1 theme to 7.2 that was built from v8.x?

**Comment by wincent** [comment]
When you say you uninstalled, do you mean you uninstalled "generator-liferay-theme" or "liferay-theme-tasks"? Either way, I think we'll need to provide a very explicit and bullet-proof instruction here on exactly what command to run.

**Comment by john-co** [comment]
I uninstalled and installed both "generator-liferay-theme" and "liferay-theme-tasks" and got the same error.
You're correct, the package.json still refers to v8 of the theme tasks.

**Comment by john-co** [comment]
Running `npm install -g liferay-theme-tasks@9.0.0-alpha.1` did not fix the issue. But when I ran the explicit command as you suggested on the generated theme root and was able to continue with upgrade process.

**Comment by wincent** [comment]
Thanks for confirming, @john-co. I'll add some hint message to the tool that makes that very explicit instead of just saying "is not supported".

**Comment by wincent** [comment]
Just released [v8.0.3](https://github.com/liferay/liferay-js-themes-toolkit/releases/tag/v8.0.3) so marking this as closed.
